Ft priority job 158174598

matthewacha opened the pull request:
#### What does this PR do?
- Add priority allocation for different jobs
- Add escalation property for every job
- Apply migration to models

#### Description of Task to be completed?
Implementing the following test cases
- api/models.py
  - Add n_nags, priority, escalate, protocol columns in check model
  - Add logic to check send_alert method to show when escalate is true
  - Add notify_escalate static method to send alerts to protocol list when escalate is true  
- transport.py
  - Edit transport classes to take two parameters

- hc/front/views.py
  - Add update_priority method to allocate priority

-front/my_checks.html
 -Add view for priority
-Run migrations

#### How should this be manually tested?
After cloning the repo, run the following commands
-set environment variables
-make
-python manage.py makemigrations
-python manage.py test

#### Any background context you want to provide?
User should be able to allocate different priorities to different health checks as per a pre-defined priority criteria

#### What are the relevant pivotal tracker stories?
[#158174598](https://www.pivotaltracker.com/n/projects/2177206)

matthewacha committed:
Permission for shell file

matthewacha committed:
[feature] allocate different priorities to different health checks
- Add column for priorities in checks table

[Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks
- Add column for priorities in checks table

[Delivers #158174598]

matthewacha committed:
Merge branch 'develop' of https://github.com/andela/hc-aces-kla into develop

matthewacha committed:
[feature]  allocate different priorities to different health checks
- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature]  allocate different priorities to different health checks
- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature]  allocate different priorities to different health checks
- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature]  allocate different priorities to different health checks
- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

- Create UI to allocate priority levels for jobs
- Write tests for priority create
- Include priority update in views
- Writ tests for adding escalation protocol
- Add escalation query

[Delivers #158174598]

matthewacha committed:
Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598

matthewacha committed:
Fix travis

matthewacha committed:
[feature] allocate different priorities to different health checks

    - Create UI to allocate priority levels for jobs
    - Write tests for priority create
    - Include priority update in views
    - Writ tests for adding escalation protocol
    - Add escalation query

    [Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

    - Create UI to allocate priority levels for jobs
    - Write tests for priority create
    - Include priority update in views
    - Writ tests for adding escalation protocol
    - Add escalation query

    [Delivers #158174598]

matthewacha committed:
Fix travis

matthewacha committed:
Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598

matthewacha committed:
Add sendgrid to requirements

matthewacha committed:
Add sendgrid to requirements

matthewacha committed:
Fix models

matthewacha committed:
[feature] allocate different priorities to different health checks

        - Create UI to allocate priority levels for jobs
        - Write tests for priority create
        - Include priority update in views
        - Write tests for adding escalation protocol
        - Add escalation
        - Write tests for protocol list
        - write notification alerts for team members in protocol

[Delivers #158174598]

matthewacha committed:
Fix models

matthewacha committed:
[feature] allocate different priorities to different health checks

        - Create UI to allocate priority levels for jobs
        - Write tests for priority create
        - Include priority update in views
        - Write tests for adding escalation protocol
        - Add escalation
        - Write tests for protocol list
        - write notification alerts for team members in protocol

[Delivers #158174598]

EdwinKato commented:
@matthewacha Resolve conflicts. Also let me know when this is finalised so that I can do a full review

matthewacha committed:
Fix merge conflict

matthewacha committed:
Fix merge conflict

matthewacha committed:
Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598

matthewacha committed:
Merge branch 'ft-priority-job-158174598' of https://github.     [feature] allocate different priorities to different health checks

            - Create UI to allocate priority levels for jobs
            - Write tests for priority create
            - Include priority update in views
            - Write tests for adding escalation protocol
            - Add escalation
            - Write tests for protocol list
            - write notification alerts for team members in protocol

    [Delivers #158174598]

matthewacha committed:
[feature] allocate different priorities to different health checks

            - Create UI to allocate priority levels for jobs
            - Write tests for priority create
            - Include priority update in views
            - Write tests for adding escalation protocol
            - Add escalation
            - Write tests for protocol list
            - write notification alerts for team members in protocol
            - Edit procfile

    [Delivers #158174598]Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598

matthewacha committed:
[feature] allocate different priorities to different health checks

            - Create UI to allocate priority levels for jobs
            - Write tests for priority create
            - Include priority update in views
            - Write tests for adding escalation protocol
            - Add escalation
            - Write tests for protocol list
            - write notification alerts for team members in protocol

    [Delivers #158174598]Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598

matthewacha committed:
Merge branch 'develop' into ft-priority-job-158174598

matthewacha committed:
Fix conflicts

matthewacha committed:
Merge branch 'develop' of https://github.com/andela/hc-aces-kla into develop

EdwinKato commented:
@matthewacha Resolve conflicts and ensure the new code follows pep8 standards

matthewacha committed:
Merge branch 'develop' into ft-priority-job-158174598

matthewacha committed:
Merge branch 'develop' of https://github.com/andela/hc-aces-kla into develop

matthewacha committed:
Implement PEP8 standards

matthewacha committed:
Implement PEP8 standards

matthewacha committed:
[feature] Fix changes from PR
- Implement PEP8 standards
- clean up

[Delivers #158174598]

EdwinKato commented on hc/api/models.py:
```diff
@@ -56,8 +56,12 @@ class Meta:
     nag_intervals = models.DurationField(default=DEFAULT_NAG_TIME)
     nag_after_time = models.DateTimeField(null=True, blank=True)
+    priority = models.IntegerField(default=1)
+    n_nags = models.IntegerField(default=0)
```
Use a more descriptive variable name. `n_nags` is ...

EdwinKato commented on hc/api/tests/test_protocols.py:
```diff
@@ -0,0 +1,67 @@
+from django.contrib.auth.hashers import make_password
+
+
+def fake_twilio_notify():
```
Any reason this is still here?

matthewacha replied:
I use it to mock the notify method in TwilioSms class. see line[21] below, since I am only testing my logic in check.send_alert().

EdwinKato replied:
Ohh I see

EdwinKato commented on hc/front/tests/test_update_priority.py:
```diff
@@ -0,0 +1,72 @@
+        self.check = Check(user=self.alice)
+        self.check.save()
+
+    def test_it_works(self):
```
Use a more descriptive method name. `test_it_works` is ambiguous

matthewacha committed:
Fix changes in naming convention

matthewacha committed:
Fix changes in naming convention

EdwinKato commented:
<img width="708" alt="screen shot 2018-07-12 at 15 24 02" src="https://user-images.githubusercontent.com/12659000/42633018-d1e32e56-85e7-11e8-8674-9b5ec9d31a59.png">

@matthewacha Fix pep8 issues

matthewacha committed:
Merge branch 'ft-priority-job-158174598' of https://github.com/andela/hc-aces-kla into ft-priority-job-158174598
